docs/migration/migrating-enact-apps.md

```diff
@@ -0,0 +1,531 @@
+
+#### `Slider.SliderTooltip`
+The boolean props `horizontal` and `vertical` have been replaced by the `orientation` prop.
+##### Example
+###### 1.x
+```
+<SliderTooltip /> // `horizontal` prop is `true` by default.
+<SliderTooltip vertical />
+```
+###### 2.0
+```
+<SliderTooltip /> // `orientation` prop is `"horizontal"` by default.
+<SliderTooltip orientation="vertical" />
+```
+
+#### `VideoPlayer`
+The `containerId` prop has changed to `spotlightId`.
+##### Example
+###### 1.x
+```
+<VideoPlayer containerId="spotlightContainer" />
+```
+###### 2.0
+```
+<VideoPlayer spotlightId="spotlightContainer" />
+```
+The `tooltipHideDelay` prop has been removed with no replacement.
+
+#### `VirtualFlexList`
+This component is replaced by `ui/VirtualFlexList`.
+
+#### `VirtualGridList`
+The `component` and `data` props are replaced by `itemRenderer`.  `itemRenderer` should
+return the component to render from a set of arbitrary items given an index.
+##### Example
+###### 1.x
+```
+...
+import Item from '@enact/moonstone/Item';
+const items = [];
+const ListItem = ({data, index}) => {
+	const {title} = data[index];
+	return (
+		<Item>
+			{title}
+		</Item>
+	);
+};
+...
+<VirtualGridList
+	...
+	component={ListItem}
+	data={items}
+	...
+/>
+```
+###### 2.0
+```
+...
+import Item from '@enact/moonstone/Item';
+const items = [];
+const itemRenderer = ({index}) => {
+	const {title} = items[index];
+	return (
+		<Item>
+			{title}
+		</Item>
+	);
+};
+...
+<VirtualGridList
+	...
+	itemRenderer={itemRenderer}
+	...
+/>
+```
+
+#### `VirtualGridList.GridListImageItem`
+This component is replaced by `ui/GridListImageItem`.
+
+#### `VirtualList`
+The `scrollTo` method's `indexToFocus` option has been removed.  Use the `focus` option and
+scroll by `index` or `node`.
+##### Example
+###### 1.x
+```
+...
+const cbScrollTo = () => {
+	this.scrollTo({indexToFocus: 1});
+};
+...
+<VirtualList cbScrollTo={cbScrollTo} ... />
+...
+```
+###### 2.0
+```
+...
+const cbScrollTo = () => {
+	this.scrollTo({focus: true, index: 1});
+};
+...
+<VirtualList cbScrollTo={cbScrollTo} ... />
+...
+```
+The `component` and `data` props are replaced by `itemRenderer`.  `itemRenderer` should return
+the component to render from a set of arbitrary items given an index.
+##### Example
+###### 1.x
+```
+...
+import Item from '@enact/moonstone/Item';
+const items = [];
+const ListItem = ({data, index}) => {
+	const {title} = data[index];
+	return (
+		<Item>
+			{title}
+		</Item>
+	);
+};
+...
+<VirtualList
+	...
+	component={ListItem}
+	data={items}
+	...
+/>
+```
+###### 2.0
+```
+...
+import Item from '@enact/moonstone/Item';
+const items = [];
+const itemRenderer = ({index}) => {
+	const {title} = items[index];
+	return (
+		<Item>
+			{title}
+		</Item>
+	);
+};
+...
+<VirtualList
+	...
+	itemRenderer={itemRenderer}
+	...
+/>
+```
+
+### spotlight
+Selectors targeting the `spotlight` container attributes `data-container-disabled` or `data-container-muted`
```

Comment:
Let's add a general note about adding support `spotlightId` to any `spotlight/Spottable` component.

Something like:
```
`spotlight/Spottable` added support for the `spotlightId` prop which can be used to identify the component in calls to `Spotlight.focus()`. In cases where you might have added a custom data attribute to identify a node, you can now use `spotlightId` instead.

1.x => <MyComponent data-component-id="my-first-component" />, Spotlight.focus('[data-component-id="my-first-component"]])
2.x => <MyComponent spotlightId="my-first-component" />, Spotlight.focus('my-first-component')
```